src/Mod/Sketcher/Gui/Utils.cpp

```diff
@@ -315,39 +315,51 @@ void SketcherGui::ActivateHandler(Gui::Document* doc, DrawSketchHandler* handler
     }
 }
 
-bool SketcherGui::isCommandActive(Gui::Document* doc, bool actsOnSelection)
-{
+bool SketcherGui::isSketchInEdit(Gui::Document* doc) {
     if (doc) {
         // checks if a Sketch Viewprovider is in Edit and is in no special mode
-        if (doc->getInEdit()
-            && doc->getInEdit()->isDerivedFrom(SketcherGui::ViewProviderSketch::getClassTypeId())) {
-            auto mode =
-                static_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit())->getSketchMode();
-            if (mode == ViewProviderSketch::STATUS_NONE
-                || mode == ViewProviderSketch::STATUS_SKETCH_UseHandler) {
-                if (!actsOnSelection)
-                    return true;
-                else if (Gui::Selection().countObjectsOfType(
-                             Sketcher::SketchObject::getClassTypeId())
-                         > 0)
-                    return true;
+        auto* vp = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+        return (vp != nullptr);
```

Comment:
```suggestion
        return (!vp);
```

Reply:
ok

Reply:
Actually I have changed my mind here after looking to the context.

First, it is very ```return (!vp)``` is equivalent to ```return (vp == nullptr)```, the equivalent to the original code would be ```return vp``` (if vp is nullptr the original code returns false, the proposed change returns true).

It is very common to check that the pointer is not nullptr before accessing it without making nullptr explicit, as in:
``` if(vp && vp->myf())```

I would not conceive writing:
```if(vp != nullptr && vp->myf()) ```

This can be done because the standard mandates a conversion of a nullptr to boolean false. It is easy to read: "if my pointer is valid and myf returns true"

When returning, I see I am returning a pointer to the vp. I may be mislead into thinking the function is returning the vp, unless I look at the return type (which if is auto, would not help me either). The context is a little bit further away. For this reason, I think it is not easier to read the original: "The pointer is not nullptr, so I am returning a boolean that is true"

I do not mind exchanging points of view about equivalent code. We always learn from the point of view of others. I think this falls under the realm of personal code style.

Reply:
Actually I was too fast and didn't pay attention my proposal reversed the boolean return 😨. 
Please ignore and keep your implementation, I fully agree with your statements.